### PR TITLE
Ci/shiva for testing

### PR DESCRIPTION
**Pull request by joshLong145**
CI changes for running local-tests against Shiva within a docker container.

See action run linked [here](https://github.com/LIT-Protocol/js-sdk/actions/runs/10184662022/job/28172516577) for new integration test steps.

We now pull release branches from a specified commit based for `Datil` based on the `DATIL_COMMIT_HASH` and `DATIL_BRANCH_NAME` which can be configured for pulling different branches for release testing.

In the current design of the container environment we are required to pull in `lit-assets` for contract deployment functionality. Due to this requirement the `GH_PAT` has been added as well as `GH_USERNAME` in order to authenticate with the Lit Protocol GitHub org.

`IPFS_API_KEY` is a required env var as it is needed for `lit-node` to function correctly.

**Note**: This PR will need follow up to change the branch / commit hash once `datil-cherry-pick` has been merged into `datil` which can be tracked here:

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -32,30 +32,68 @@ jobs:
...
-  e2e-connection:
+  integration-tests:
     runs-on: ubuntu-latest
     timeout-minutes: 30
     env:
-      NETWORK: datil-dev
-      DEBUG: true
-      MAX_ATTEMPTS: 3
-      CHUNK_SIZE: 9
+      DATIL_COMMIT_HASH: b5cfe7489275dbed5e9098818d0bd441c8918903
```
Comment (Ansonhkg) [inline .github/workflows/ci.yml:39]: Since we have now merged the lit-assets PR should we point this to develop?

Comment (joshLong145) [reply .github/workflows/ci.yml:39]: We will want to point it at `datil` for now as that is our current compatibility with `v7` as of now. But once we merge https://github.com/LIT-Protocol/lit-assets/pull/1341 we can switch from the branch used to [datil](https://github.com/LIT-Protocol/lit-assets/tree/datil).


I think once we start on changes for v7 like [this](https://github.com/LIT-Protocol/lit-assets/pull/1372) for example we can use `develop` in a parallel job.

**Review by Ansonhkg — APPROVED** [review]
This LGTM!

File: local-tests/setup/tinny-environment.ts [hunk 2]
```diff
@@ -20,6 +21,7 @@ import { createSiweMessage, generateAuthSig } from '@lit-protocol/auth-helpers';
 import { ShivaClient, TestnetClient } from './shiva-client';
 import { toErrorWithMessage } from './tinny-utils';
 import { CENTRALISATION_BY_NETWORK } from '@lit-protocol/constants';
+import { Network } from '../../packages/wrapped-keys/src/lib/types';
```
[the remark this answers is not in the document]
Comment (joshLong145) [reply local-tests/setup/tinny-environment.ts:24]: Good catch Will update this import

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -32,30 +32,68 @@ jobs:
...
+      DATIL_BRANCH_NAME: datil
```
[the remark this answers is not in the document]
Comment (joshLong145) [reply .github/workflows/ci.yml:40]: This is fair, the hash should in theory always be the HEAD of the branch but this is hard to enforce. going to try and pull the branch from the hash as well for consistency.

**Review by MaximusHaximus — APPROVED** [review]
LGTM :+1: Just 1 apparently unused import

File: local-tests/setup/tinny-environment.ts [hunk 4]
```diff
@@ -9,6 +9,7 @@ import { LitContracts } from '@lit-protocol/contracts-sdk';
 import {
   AuthSig,
   CosmosAuthSig,
+  LIT_NETWORKS_KEYS,
```
[the remark this answers is not in the document]
Comment (joshLong145) [reply local-tests/setup/tinny-environment.ts:12]: Nice catch was left over from a refactor.

**Review by FedericoAmura — APPROVED** [review]

**Pull request merged**